ega-data-api-res/src/main/java/eu/elixir/ega/ebi/reencryptionmvc/service/internal/CacheResServiceImpl.java

```diff
@@ -296,80 +288,50 @@ else if (sourceFormat.equalsIgnoreCase("symmetricgpg")) {
+                key = id + "_" + startPage;
+                byte[] page = pageDowloader.downloadPage(key);
                 errorLocation = 4;
-                if (get == null)
-                    throw new GeneralStreamingException(sessionId + "Error getting Cache Page " + key + " at Stage ", 8);
-                // Prefetch Loop (prefetch next few pages in background)
-                for (int prefetch = 1; prefetch <= MAX_CONCURRENT; prefetch++) {
-                    if ((cachePage + prefetch) < cacheEndPage) { // no paged past end of file
-                        final String key__ = id + "_" + (cachePage + prefetch);
-                        if (!myPageCache.containsKey(key__) && !concurrentHashSet.contains(key__)) {
-                            concurrentHashSet.add(key__);
-                            //Executors.newSingleThreadExecutor().execute(new Runnable() {
-                            Executors.newCachedThreadPool().execute(new Runnable() {
-                                @Override
-                                public void run() {
-                                    byte[] page = myPageCache.get(key__).getPage();
-                                    concurrentHashSet.remove(key__);
-                                }
-                            });
-                        }
-                    }
-                }
-                errorLocation = 5;
+                if (page == null)
+                    throw new GeneralStreamingException(sessionId + " Error getting page " + key);
 
-                ByteArrayInputStream bais = new ByteArrayInputStream(get); // Wrap byte array
-                bais.skip(cachePageOffset); // first cache page
+                ByteArrayInputStream bais = new ByteArrayInputStream(page);
+                bais.skip(pageOffset); // first cache page
 
                 // At this point the plain data is in a cache page
-                InputStream in;
+                
                 long delta = bytesToTransfer - bytesTransferred;
-                if (delta < get.length) {
+                if (delta < page.length) {
                     in = ByteStreams.limit(bais, delta);
                 } else {
                     in = bais;
                 }
-                cachePageOffset = 0;
+                pageOffset = 0;
 
                 // Copy the specified contents - decrypting through input, encrypting through output
                 long bytes = ByteStreams.copy(in, eOut);
                 errorLocation = 6;
                 bytesTransferred += bytes;
-                cachePage += 1;
+                startPage += 1;
             }
             return bytesTransferred;
         } catch (Exception ex) {
-            throw new GeneralStreamingException(sessionId + "Error Location: " + errorLocation + "\n" + ex.toString(), 10);
+            log.error(sessionId + " Error Location: " + errorLocation + "\n" + ex.toString());
```

Comment:
And change "Error location" + errorLocation for a some meaningful name please.

Reply:
@afoix Could you pls elaborate more on how do you want it?

Reply:
@jorizci can give you an example - we talk yesterday in the meeting and Jorge give you the example of the error that we are seeing in the logs is "Error Location: 3", "Error Location: 5" and this is a meaningless exception. Anyway, this is a small thing so we can do it in another moment or after to put in production the cache remove,